### PR TITLE
update the 020 solution in C++

### PR DESCRIPTION
**Pull request by Tina66**

### CODE REVIEW

File: src/020._valid_parentheses.md [hunk 1]
```diff
@@ -37,7 +39,7 @@ Example 5:
     Output: true
 ```
 
-### Solutions
+### Python Solutions
```
Comment (jovan-wong) [inline src/020._valid_parentheses.md]: Remove the space

File: src/020._valid_parentheses.md [hunk 2]
```diff
@@ -67,4 +69,40 @@ class Solution(object):
...
+
+### C++ Solutions
+
+* Time complexity : O(n)
+* Space complexity : O(n)
+
+```C++
+class Solution {
+public:
+    bool isValid(string s) {
```
Comment (jovan-wong) [inline src/020._valid_parentheses.md]: const string &s

File: src/020._valid_parentheses.md [hunk 3]
```diff
@@ -67,4 +69,40 @@ class Solution(object):
...
+        string left = "([{";
+        string right = ")]}";       
+        stack<char> stk;
+        
+        for (auto c:s) {
```
Comment (jovan-wong) [inline src/020._valid_parentheses.md]: for (auto c : s) {

File: src/020._valid_parentheses.md [hunk 4]
```diff
@@ -67,4 +69,40 @@ class Solution(object):
...
+            if (left.find(c) != string::npos) {
+                stk.push(c);
+            } else {
+                if (stk.empty() || stk.top() != left[right.find(c)])
```
Comment (jovan-wong) [inline src/020._valid_parentheses.md]:
```c++
                if (stk.empty() || stk.top() != left[right.find(c)]) {
                    return false;
                } else {
                    stk.pop();
                }
```

File: src/020._valid_parentheses.md (hunk 2 above)
Comment (Tina66) [reply src/020._valid_parentheses.md]: I understand your meaning,but this is what the problem shows. I just copy the function name and write the function body.

Comment (jovan-wong) [reply src/020._valid_parentheses.md]: OK, I see.

File: src/020._valid_parentheses.md (hunk 4 above)
Comment (Tina66) [reply src/020._valid_parentheses.md]: I have corrected some format. Thank  for your careful watching! @jovan-wong

**Comment by Tina66** [comment]
Thank u for your careful review! I will  merge this pr.

**Comment by Tina66** [comment]
Thank u for your careful review! I will merge this pr. @jovan-wong

**Pull request merged**